sql/core/src/main/scala/org/apache/spark/sql/sources/interfaces.scala

```diff
@@ -111,13 +111,11 @@ trait SchemaRelationProvider {
 }
 
 /**
- * ::Experimental::
  * Implemented by objects that can produce a streaming `Source` for a specific format or system.
  *
  * @since 2.0.0
  */
-@Experimental
-@Unstable
+@Stable
 trait StreamSourceProvider {
```

Comment:
If this is stable, I think `Source` and `Sink` need to be stable as well?

Reply:
[Source](https://github.com/apache/spark/blob/master/sql/core/src/main/scala/org/apache/spark/sql/execution/streaming/Source.scala) and [Sink](https://github.com/apache/spark/blob/master/sql/core/src/main/scala/org/apache/spark/sql/execution/streaming/Sink.scala) do not have any annotations on them. Does that imply they are stable? Sink and Source are exposed/returned in the [DataSource](https://github.com/apache/spark/blob/master/sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/DataSource.scala#L252-L301) class today as well. 

I am happy to add a Stable annotation if that is your recommendation.